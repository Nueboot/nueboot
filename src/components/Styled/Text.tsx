import styled from 'styled-components';

const Text = styled.span`
  font-family: 'Heebo', -apple-system, BlinkMacSystemFont, 'Helvetica',
    sans-serif;
  font-size: 0.875rem;
  color: rgba(0, 0, 0, 0.7);
`;

export default Text;
