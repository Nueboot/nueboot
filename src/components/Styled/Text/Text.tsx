import styled from 'styled-components';

interface TextProps {
  readonly bold?: boolean;
}

const Text = styled.span`
  font-size: 1rem;
  color: rgba( 0, 0, 0, .9 );
  font-weight: ${(props: TextProps) => props.bold ? 700 : 400};
`;

export default Text;
