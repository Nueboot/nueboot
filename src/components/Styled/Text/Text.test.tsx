import { RouteComponentProps } from '@reach/router';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Text from './Text';

describe('Text', () => {
  let component: ShallowWrapper<RouteComponentProps>;

  describe('without bold', () => {
    beforeEach(() => {
      component = shallow(<Text />);
    });

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with bold', () => {
    beforeEach(() => {
      component = shallow(<Text bold />);
    });

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
