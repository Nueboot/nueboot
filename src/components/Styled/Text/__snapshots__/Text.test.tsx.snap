// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text with bold renders correctly 1`] = `
<span
  className="sc-bdVaJa cxzrWC"
/>
`;

exports[`Text without bold renders correctly 1`] = `
<span
  className="sc-bdVaJa fKKtts"
/>
`;
